# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../kiwi/fix_ProxyFileChooserButton.patch
# Copyright (C) 2006 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- ./kiwi/ui/widgets/filechooser.py.orig	2006-05-01 00:58:57.000000000 +0200
+++ ./kiwi/ui/widgets/filechooser.py	2006-05-01 01:06:58.000000000 +0200
@@ -50,21 +50,19 @@
 class ProxyFileChooserButton(_FileChooserMixin, PropertyObject,
                              gtk.FileChooserButton, ProxyWidgetMixin):
     __gtype_name__ = 'ProxyFileChooserButton'
-    def __init__(self, title=None, parent=None,
-                 action=gtk.FILE_CHOOSER_ACTION_OPEN,
-                 buttons=None, backend=None):
+    def __init__(self, title, backend=None):
         """
         @param title:
-        @param parent:
-        @param action:
-        @param buttons:
         @param backend:
         """
         ProxyWidgetMixin.__init__(self)
         PropertyObject.__init__(self, data_type=str)
-        gtk.FileChooserButton.__init__(self, title=title,
-                                       parent=parent, action=action,
-                                       buttons=buttons, backend=backend)
+
+        # Broken, Broken PyGTK
+        if isinstance(title, str):
+            gtk.FileChooserButton.__init__(self, title, backend)
+        else:
+            gtk.FileChooserButton.__init__(self, title)
 
 class ProxyFileChooserWidget(_FileChooserMixin, PropertyObject,
                              gtk.FileChooserWidget, ProxyWidgetMixin):
